hazelcast/src/main/java/com/hazelcast/core/Cluster.java

```diff
@@ -169,6 +170,31 @@
     void changeClusterState(ClusterState newState, TransactionOptions transactionOptions);
 
     /**
+     * The cluster version indicates the operating version of the cluster. It is separate from each node's codebase version,
+     * as it may be acceptable for a node to operate at a different compatibility version than its codebase version. This method
+     * may return {@code null} if invoked after the ClusterService is constructed and before the node forms a cluster, either
+     * by joining existing members or becoming master of its standalone cluster if it is the first node on the cluster.
```

Comment:
OK, as discussed there will be a property to set the version that should be used before the proper cluster version is set. It's for the sake of backward compatibility, to properly serialize the join messages.